### PR TITLE
Infer .quilt on s3 bucket registries

### PR DESCRIPTION
**Pull request by meffij**

### CODE REVIEW

File: api/python/t4/packages.py [hunk 1]
```diff
@@ -136,6 +136,14 @@ def __init__(self, name=None, pkg_hash=None, registry=None):
             # default to local registry
             registry = get_package_registry()
 
+        parsed_registry = urlparse(registry)
+        if parsed_registry.scheme == 's3' and parsed_registry.path in ['', '/']:
```
Comment (stevededalus) [inline api/python/t4/packages.py]: shouldn't this logic be in the get_registry function so its consistent with push and build?

Comment (meffij) [reply api/python/t4/packages.py]: no, since this is an override of a user-specified registry rather than an inferred registry from a path

Comment (stevededalus) [reply api/python/t4/packages.py]: but isn't this what `get_registry(path)` does? It takes a path determines if its s3 and the returns the bucket + `/.quilt`

Comment (stevededalus) [inline api/python/t4/packages.py]: list_packages also uses a user supplied registry.

Comment (meffij) [reply api/python/t4/packages.py]: I think you're right, we may need to split the code eventually but for now we can use `get_registry`

**Pull request merged**